api/models/Centres.js

```diff
@@ -190,8 +190,13 @@ const model = {
   },
 
   afterDestroy: function (records, done) {
-    _.map(records, (record) => this.publishDestroy(record.id, record));
-    done();
+    Promise.all(_.map(records, (record) =>
+        Movement.destroy({ centre: record.id })
+          .then(() => Prebooking.destroy({ centre: record.id }))
+          .then(() => Event.destroy({ centre: record.id }))
+          .then(() => this.publishDestroy(record.id, record))
+      ))
+      .then(() => done());
   },
```

Comment:
Added a destroy for the `detainee`
We can't (easily) do a `subject` since they can be referred to by multiple movement orders